packages/grid/_modules_/grid/components/columnHeaders/ColumnHeadersItemCollection.tsx

```diff
@@ -35,19 +35,22 @@ export function ColumnHeaderItemCollection(props: ColumnHeadersItemCollectionPro
   // TODO refactor by putting resizing in the state so we avoid adding listeners.
   useApiEventHandler(apiRef!, COL_RESIZE_START, handleResizeStart);
   useApiEventHandler(apiRef!, COL_RESIZE_STOP, handleResizeStop);
-
-  const items = columns.map((col, idx) => (
-    <ColumnHeaderItem
-      key={col.field}
-      {...sortColumnLookup[col.field]}
-      filterItemsCounter={filterColumnLookup[col.field] && filterColumnLookup[col.field].length}
-      options={options}
-      isDragging={col.field === dragCol}
-      column={col}
-      colIndex={idx}
-      isResizing={resizingColField === col.field}
-    />
-  ));
+  const items = columns.map((col, idx) => {
+    const isLastColumn = idx === columns.length - 1;
+    return (
+      <ColumnHeaderItem
+        key={col.field}
+        {...sortColumnLookup[col.field]}
+        filterItemsCounter={filterColumnLookup[col.field] && filterColumnLookup[col.field].length}
+        options={options}
+        isDragging={col.field === dragCol}
+        column={col}
+        colIndex={idx}
+        isLastColumn={isLastColumn}
```

Comment:
A detail coding style convention, we generally don't use `is` for boolean.

```suggestion
        lastColumn={lastColumn}
```